.github/workflows/dependency-check.yaml

```
@@ -0,0 +1,58 @@
+name: MacawUI migration
+on: [pull_request]
+
+jobs:
+  current_state:
+    runs-on: ubuntu-latest
+    name: "Calculate current state"
+    outputs: 
+      main_statistics: ${{ steps.store_import_stats.outputs.MAIN_STATISTICS }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: "main"
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - name: Install deps
+        run: |
+          npm ci
+      - name: Store import stats
+        id: store_import_stats
+        run: |
+          stats=$(npx depcruise --config .dependency-cruiser.js src --output-type json | jq ".summary.info")
+          echo "Statistics on main branch: ${stats}"
+          echo "MAIN_STATISTICS=${stats}" >> $GITHUB_OUTPUT
+
+  check_migation_status:
+    runs-on: ubuntu-latest
+    name: "Migration check"
+    needs: current_state
+    steps:
+      - uses: actions/checkout@v2
```

Consistency and removal of deprecated components:

```suggestion
      - uses: actions/checkout@v3
```

---

thanks!